detox/src/devices/drivers/EmulatorDriver.js

```diff
@@ -1,104 +1,89 @@
-        throw new Error(`Emulator with name ${name} has a corrupt config.ini file (${configFile}), try fixing it by recreating an emulator.`);
-      }
-
-      config['skin.name'] = `${width}x${height}`;
-      fs.writeFileSync(configFile, ini.stringify(config));
-    }
-    return config;
+  name() {
+    return this._name
   }
 
-  async boot(avdName) {
-    let adbName = await this._findADBNameByAVDName(avdName, { strict: false });
-    const coldBoot = adbName == null;
+  async acquireFreeDevice(avdName) {
+    await this._validateAvd(avdName);
+    await this._fixEmulatorConfigIniSkinNameIfNeeded(avdName);
 
-    // If it's not already running, start it now.
-    if (coldBoot) {
-      await this.emulator.boot(avdName);
-      adbName = await this._findADBNameByAVDName(avdName, { strict: true });
-    }
+    const adbName = await this.deviceRegistry.getDevice(avdName);
+    await this._boot(avdName, adbName);
 
-    await this._waitForBootToComplete(adbName);
-    await this.emitter.emit('bootDevice', { coldBoot, deviceId: adbName });
+    await this.adb.apiLevel(adbName);
+    await this.adb.unlockScreen(adbName);
 
+    this._name = `${adbName} (${avdName})`;
     return adbName;
   }
 
-  async _findADBNameByAVDName(avdName, { strict }) {
-    const adbDevices = await this.adb.devices();
-    const filteredDevices = _.filter(adbDevices, {type: 'emulator', name: avdName});
-
-    if (filteredDevices.length === 1) {
-      return filteredDevices[0].adbName;
-    }
+  async cleanup(adbName, bundleId) {
+    await this.deviceRegistry.freeDevice(adbName);
+    await super.cleanup(adbName, bundleId);
+  }
 
-    if (filteredDevices.length > 1) {
-      throw new Error(`Got more than one device corresponding to the name: ${avdName}`);
-    }
+  async _boot(avdName, adbName) {
+    const coldBoot = !!this.pendingBoots[adbName];
 
-    if (strict) {
-      throw new Error(`Could not find '${avdName}' on the currently ADB attached devices, 
-      try restarting adb 'adb kill-server && adb start-server'`);
+    if (coldBoot) {
+      const port = this.pendingBoots[adbName];
+      await this.emulator.boot(avdName, {port});
+      delete this.pendingBoots[adbName];
     }
 
-    return null;
+    await this._waitForBootToComplete(adbName);
+    await this.emitter.emit('bootDevice', { coldBoot, deviceId: adbName });
   }
 
-  async acquireFreeDevice(avdName) {
+  async _validateAvd(avdName) {
     const avds = await this.emulator.listAvds();
-
     if (!avds) {
-      const avdmanagerPath = path.join(Environment.getAndroidSDKPath(), 'tools', 'bin', 'avdmanager');
+      const avdmanagerPath = path.join(environment.getAndroidSDKPath(), 'tools', 'bin', 'avdmanager');
 
       throw new Error(`Could not find any configured Android Emulator. 
-      Try creating a device first, example: ${avdmanagerPath} create avd --force --name Nexus_5X_API_24 --abi x86 --package 'system-images;android-24;google_apis_playstore;x86' --device "Nexus 5X"
```

Comment:
no `Pixel 2` target anymore, only `pixel`

Reply:
Not sure I have understood — you have changed the code to use `Pixel 2` target, no?